jujugui/static/gui/src/app/app.js

```diff
@@ -1708,90 +1685,139 @@ YUI.add('juju-gui', function(Y) {
-          modelList.forEach(model => {
-            const path = `${model.owner}/${model.name}`;
-            // Don't need to store the path if it already exists.
-            if (!this.userPaths.get(path)) {
-              this.userPaths.set(path, {
-                type: 'model',
-                model: model
-              });
+    _fetchEntityFromUserState: function(userState) {
+      const legacyPath =
+        window.jujulib.URL.fromString('u/' + userState).legacyPath();
+      const userPaths = this.userPaths;
+      const entityCache = userPaths.get(userState);
+      if (entityCache && entityCache.promise) {
+        return entityCache.promise;
+      }
+      const entityPromise = new Promise((resolve, reject) => {
+        this.get('charmstore').getEntity(
+          legacyPath, (err, entityData) => {
+            if (err) {
+              console.error(err);
+              reject(userState);
+              return;
             }
+            resolve(userState);
           });
-          // Attempt to handle the path.
-          if (!handleEntity()) {
-            // If the path does not exist after we've populated the list with
-            // models then it must be for a charm/bundle so store it as such.
-            this.userPaths.set(userPath, {type: 'store'});
-            // Handle the entity. It will exist now.
-            handleEntity();
-          }
+      });
+      userPaths.set(userState, {promise:entityPromise});
+      return entityPromise;
+    },
+
+    _listAndSwitchModel: function(modelName, modelUUID) {
+      this.controllerAPI.listModelsWithInfo((err, modelList) => {
+        if (err) {
+          console.error('unable to list models', err);
+          this.db.notifications.add({
+            title: 'Unable to list models',
+            message: 'Unable to list models: ' + err,
+            level: 'error'
+          });
+          return;
+        }
+        const noErrorModels = modelList.filter(model => !model.err);
+        let model = undefined;
+        if (modelName) {
+          model = noErrorModels.find(model => model.name === modelName);
+        } else if (modelUUID) {
+          model = noErrorModels.find(model => model.uuid === modelUUID);
+        }
+        const rootUserName = this._getAuth().rootUserName;
+        if (model) {
+          this.maskVisibility(false);
+          this.state.changeState({
+            model: {
+              path: `${rootUserName}/${model.name}`,
+              uuid: model.uuid
+            },
+            user: null,
+            root: null
+          });
+        } else {
+          // If no model was found then put the user in unconnected mode
+          this.state.changeState({
+            root: null,
+            user: rootUserName
+          });
+        }
+      });
+    },
+
+    _disambiguateUserState: function(entityPromise) {
+      entityPromise.then(userState => {
+        console.log('entity found, showing store');
```

Comment:
This isn't really the store, is it? It's a details page, which isn't quite the same thing. No /store path, for example.